### PR TITLE
Push Notification

### PR DESCRIPTION
**Pull request by RyanRory**
Fetch data when receiving a silent notification.

### First time contributor checklist

- [ ] I have read the [README](https://github.com/WhisperSystems/Signal-iOS/blob/master/README.md) and [CONTRIBUTING](https://github.com/WhisperSystems/Signal-iOS/blob/master/CONTRIBUTING.md) documents
- [ ] I have signed the [Contributor Licence Agreement](https://whispersystems.org/cla/)

### Contributor checklist

- [ ] I'm following the [code, UI and style conventions](https://github.com/WhisperSystems/Signal-iOS/blob/master/CONTRIBUTING.md#code-conventions)
- [ ] My commits are rebased on the latest master branch
- [ ] My commits are in nice logical chunks
- [ ] My contribution is fully baked and is ready to be merged as is
- [ ] I have tested my contribution on these devices:
 * iDevice A, iOS X.Y.Z
 * iDevice B, iOS Z.Y

- - - - - - - - - -

### CODE REVIEW

File: Signal/src/AppDelegate.m [hunk 1]
```diff
@@ -1094,6 +1095,17 @@ - (void)application:(UIApplication *)application didReceiveRemoteNotification:(N
         OWSLogInfo(@"Ignoring remote notification; app not ready.");
         return;
     }
+    
+    //deal with remote notification
+    //fetch data
+    NSLog(@"Receive remote notification!");
+    __block AnyPromise *job = [AppEnvironment.shared.messageFetcherJob run].then(^{
```
Comment (Mikunj) [inline Signal/src/AppDelegate.m]: we can just simplify this to
```
[[AppEnvironment.shared.messageFetcherJob run] retainUntilComplete];
```

File: Signal/src/AppDelegate.m [hunk 2]
```diff
@@ -1109,6 +1121,19 @@ - (void)application:(UIApplication *)application
         OWSLogInfo(@"Ignoring remote notification; app not ready.");
         return;
     }
+
+    //deal with remote notification
+    //fetch data
+    NSLog(@"Receive remote notification!");
+    __block AnyPromise *job = [AppEnvironment.shared.messageFetcherJob run].then(^{
```
Comment (Mikunj) [inline Signal/src/AppDelegate.m]: Same with this one

**Comment by gmbnt** [comment]
`PushNotificationManager` would make more sense in SignalServiceKit

**Comment by gmbnt** [comment]
Made some minor changes here: https://github.com/loki-project/loki-messenger-ios/commit/86a9e6534ed04d7913b7a8b1520b55b32689c09f

**Pull request merged**